laterpay/application/Controller/Admin/TinyMCE.php

```diff
@@ -0,0 +1,196 @@
+                array( 'add_tinymce_button' ),
+            ),
+            'laterpay_admin_enqueue_styles_post_edit' => array(
+                array( 'laterpay_on_admin_view', 200 ),
+                array( 'localize_script', 50 ),
+            ),
+            'laterpay_admin_enqueue_styles_post_new'  => array(
+                array( 'laterpay_on_admin_view', 200 ),
+                array( 'localize_script', 50 ),
+            ),
+        );
+
+    }
+
+    /**
+     * To add laterpay short code generator dropdown in TinyMCE.
+     *
+     * @param LaterPay_Core_Event $event
+     *
+     * @return void
+     */
+    public function register_tinymce_button( LaterPay_Core_Event $event ) {
+
+        $result = $event->get_result();
+
+        $result = array_merge(
+            $result,
+            [
+                'laterpay_shortcode_generator',
+            ]
+        );
+
+        $event->set_result( $result );
+
+    }
+
+    /**
+     * To register script for custom button for shortcode generator dropdown.
+     *
+     * @param LaterPay_Core_Event $event
+     *
+     * @return void
+     */
+    public function add_tinymce_button( LaterPay_Core_Event $event ) {
+
+        $result = $event->get_result();
+
+        $result['laterpay_shortcode_generator'] = sprintf( '%slaterpay-backend-shortcode-generator.js', $this->config->js_url );
+
+        $event->set_result( $result );
+    }
+
+    /**
+     * To localize text for shortcode generator.
+     *
+     * @return void
+     */
+    public function localize_script() {
+
+        // Time passes and vouchers data.
+        $time_passes_model = LaterPay_Model_TimePassWP::get_instance();
+        $time_passes_list  = $time_passes_model->get_active_time_passes();
+        $time_passes_ids   = array();
+
+        foreach ( $time_passes_list as $item ) {
+
+            if ( empty( $item['title'] ) || empty( $item['pass_id'] ) ) {
+                continue;
+            }
+
+            $time_passes_ids[] = array(
+                'text'  => $item['title'],
+                'value' => $item['pass_id'],
+            );
+        }
+
+        // Subscriptions data.
+        $subscriptions_model = LaterPay_Model_SubscriptionWP::get_instance();
+        $subscriptions_list  = $subscriptions_model->get_active_subscriptions();
+        $subscriptions_ids   = [];
+
+        foreach ( $subscriptions_list as $item ) {
+
+            if ( empty( $item['title'] ) || empty( $item['id'] ) ) {
+                continue;
+            }
+
+            $subscriptions_ids[] = array(
+                'text'  => $item['title'],
+                'value' => $item['id'],
+            );
+        }
+
+        wp_localize_script(
+            'laterpay-post-edit',
+            'laterpay_shortcode_generator',
+            array(
+                'button'                       => array(
+                    'text' => esc_html__( 'LaterPay ShortCodes', 'laterpay' ),
```

Comment:
Would be possible to use an icon instead?

Reply:
Icon added.